### ISSUE TITLE
How to use LSTM layer in tfLite ?

### ISSUE THREAD

**Issue by jiarenyf** [tensorflow/tensorflow#20881]
How to use the following code ?

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/kernels/lstm.cc

**Comment by aselle** [comment]
This is currently not easy to do. We are working on creating a simple example. Could you give some more details on your exact needs?

**Comment by jiarenyf** [comment]
I am looking for a way to deploy the `CNN+stack_bidirectional_dynamic_rnn+CTC` on mobile devices, and it seems that LSTM layers are not supported by tf-lite, but I happened to find the following code, so I want to know is it possible to support LSTM layers with tf-lite, thank you ?

[](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/kernels/lstm.cc)

**Comment by aselle** [comment]
LSTM's are supported as operations, but it is currently difficult to generate such models automatically. We will keep you posted when we have an easy path to support this.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by david456314** [comment]
@aselle  I am also concerned about this, is it mean that LSTM structure is not supported in tf-lite now or when can it be supported in the future?
Can we  implement it by compose some OPs currently?

**Comment by zhangjinhong17** [comment]
@aselle , I want to run LSTM in tflite also.
When can it be supported?

**Comment by jiarenyf** [comment]
...

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by samwit** [comment]
Any update on when a simple way to do this is coming? Will this convert tf.keras LSTMs?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by kvcuong** [comment]
Hi, 

I would like to know if tflite at this time support RNN (LSTM, Multi-dim LSTM) ?

**Comment by sharypovandrey** [comment]
I also trying to convert keras biLSTM to tflite model, but still have the same issues. Waiting for a support
